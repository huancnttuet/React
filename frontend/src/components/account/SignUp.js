import React, { useState } from 'react'
import {
	Form,
	Button,
	Col,
	Row,
	Container,
	Alert,
	Spinner
} from 'react-bootstrap'
import axios from 'axios'
import TopPage from '../../views/TopPage'
import PopUp from '../utils/PopUp'

function SignUp(props) {
	const emailSignUp = useFormInput('')
	const usernameSignUp = useFormInput('')
	const [loading, setLoading] = useState(false)
	const [stateSignUp, setStateSignUp] = useState('')
	const [success, setSuccess] = useState(false)
	const [popUp, setPopUp] = useState(false)

	function handleClick() {
		if (!checkUsername()) {
			setStateSignUp('username cần có ít nhất 6 ký tự')
		} else {
			var data = {
				emailSignUp: emailSignUp.value,
				usernameSignUp: usernameSignUp.value
			}
			setLoading(true)
			axios.post('http://localhost:8080/users/signup', { data }).then((res) => {
				setLoading(false)
				console.log(res.data)
				setPopUp(true)
				if (res.data.status === 'Success') {
					setSuccess(true)
				} else {
					setSuccess(false)
				}
				setStateSignUp(res.data.message)
			})
		}
	}

	function checkUsername() {
		if (usernameSignUp.value.length < 6) return false
		return true
	}

	return (
		<div>
			<Container>
				<Row style={{ marginTop: 50 }}>
					<Col></Col>
					<Col>
						{loading ? (
							<Spinner animation='border' role='status'></Spinner>
						) : (
							<Form>
								<Form.Row>
									<Form.Group as={Col} controlId='emailSignUp' {...emailSignUp}>
										<Form.Label>Email</Form.Label>
										<Form.Control type='email' placeholder='Enter email' />
									</Form.Group>
								</Form.Row>
								<Form.Row>
									<Form.Group
										as={Col}
										controlId='usernameSignUp'
										{...usernameSignUp}
									>
										<Form.Label>Username</Form.Label>
										<Form.Control placeholder='Enter username' />
									</Form.Group>
								</Form.Row>

								{stateSignUp ? (
									<Alert variant='warning' style={{ fontSize: '14px' }}>
										{stateSignUp}
									</Alert>
								) : (
									''
								)}

								<PopUp
									show={popUp}
									onHide={() => setPopUp(false)}
									message={stateSignUp}
									title={
										success
											? 'Tạo tài khoản thành công'
											: 'Tạo tài khoản thất bại'
									}
								/>

								<Form.Group id='formGridCheckbox'>
									<Form.Check type='checkbox' label='Check me out' />
								</Form.Group>

								<Button id='submitBtn' variant='primary' onClick={handleClick}>
									Submit
								</Button>
							</Form>
						)}
					</Col>
					<Col></Col>
				</Row>
			</Container>
		</div>
	)
}

function useFormInput(initial) {
	const [value, setValue] = useState(initial)

	function handleChange(e) {
		setValue(e.target.value)
	}
	return {
		value,
		onChange: handleChange
	}
}

export default SignUp
