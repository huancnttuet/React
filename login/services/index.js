export { default as serverServices } from "./user";
