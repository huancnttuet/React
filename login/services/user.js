var user = require("../model/user.js");
var nodemailer = require("nodemailer");

module.exports = {
  signin: async (usernameSignIn, pwdSignIn) => {
    var valueSignIn = await user.checkSignIn(usernameSignIn, pwdSignIn);

    if (valueSignIn !== 0) {
      console.log("login true");
      return {
        code: "SUCCESS",
        message: "Đăng nhập thành công",
        result: {
          username: valueSignIn.username,
          id: valueSignIn.id
        }
      };
    } else {
      console.log("login false");
      return {
        code: "ERROR",
        message: "Sai mật khẩu"
      };
    }
  },
  signup: async (emailSignUp, usernameSignUp) => {
    const sendMail = pwd => {
      var transporter = nodemailer.createTransport({
        service: "Gmail",
        auth: {
          user: "huancnttuet@gmail.com",
          pass: "341997mok"
        }
      });

      var mailOptions = {
        from: "huancnttuet@gmail.com",
        to: "huancnttmta@gmail.com",
        subject: "Password demo",
        text: `Yourpassword: ${pwd}`
      };

      transporter.sendMail(mailOptions, function(error, info) {
        if (error) {
          console.log(error);
        } else {
          console.log("Email sent: " + info.response);
        }
      });
    };
    const createPwdSignUp = () => {
      var length = 8,
        charset =
          "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789",
        retVal = "";
      for (var i = 0, n = charset.length; i < length; ++i) {
        retVal += charset.charAt(Math.floor(Math.random() * n));
      }
      return retVal;
    };

    var checkSU = await user.checkSignUp(emailSignUp, usernameSignUp);
    console.log(checkSU);
    if (!checkSU) {
      var pwdSignUp = createPwdSignUp();

      if (await user.createUser(emailSignUp, usernameSignUp, pwdSignUp)) {
        sendMail(pwdSignUp);
        return {
          code: "SUCCESS",
          message: "create new user success"
        };
      } else {
        return {
          code: "ERROR",
          message: "error create new user"
        };
      }
    } else {
      return {
        code: "EXIST",
        message: "user or mail exist"
      };
    }
  },
  changepwd: async (id, pwd, pwdNew) => {
    if (await user.checkIdPwd(id, pwd)) {
      if (await user.changePwd(id, pwdNew)) {
        return {
          code: "SUCCESS",
          message: "Đổi mật khẩu thành công"
        };
      } else {
        return {
          code: "ERROR",
          message: "Đã có lỗi xảy ra"
        };
      }
    } else {
      return {
        code: "SAIPASS",
        message: "Sai mật khẩu"
      };
    }
  },
  forgottenacc: async emailFA => {
    var pwd = await user.checkEmailFA(emailFA);
    const sendMail = pwd => {
      var transporter = nodemailer.createTransport({
        service: "Gmail",
        auth: {
          user: "huancnttuet@gmail.com",
          pass: "341997mok"
        }
      });

      var mailOptions = {
        from: "huancnttuet@gmail.com",
        to: "huancnttmta@gmail.com",
        subject: "Forgotten Password demo",
        text: `Yourpassword: ${pwd}`
      };

      transporter.sendMail(mailOptions, function(error, info) {
        if (error) {
          console.log(error);
          return {
            code: "ERROR",
            message: "Lỗi trong quá trình gửi mail"
          };
        } else {
          console.log("Email sent: Quen pass " + info.response);
          return {
            code: "SUCCESS",
            message: "Pass đã được gửi lại về mail của bạn"
          };
        }
      });
    };
    if (pwd !== null) {
      sendMail(pwd);
    } else {
      return {
        code: "NOTFOUND",
        message: "Mail chưa được đăng ký"
      };
    }
  }
};
