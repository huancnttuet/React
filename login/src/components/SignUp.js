import React, {useState} from 'react';
import {Form, Button, Col, Row, Container} from 'react-bootstrap'
import axios from 'axios'


function SignUp(props) {
  const emailSignUp = useFormInput('')
  const usernameSignUp = useFormInput('')

  const [stateSignUp, setStateSignUp] = useState('')

  function handleClick() {

    if(!checkUsername()) {
      setStateSignUp('errorUsername')
    } else {
      var data = {
        emailSignUp: emailSignUp.value,
        usernameSignUp: usernameSignUp.value
      }
      axios.post('http://localhost:8000/signup', {data}).then( (res) => {
        setStateSignUp(res.data.message)
      })
    }
  }

  function checkUsername() {
    if(usernameSignUp.value.length < 6)
     return false
    return true
  }

  return(
    <div>

      <Container>
        <Row style={{marginTop:50}}>
          <Col></Col>
          <Col>
            <Form>
            <Form.Row>
              <Form.Group as={Col} controlId="emailSignUp" {...emailSignUp}>
                <Form.Label>Email</Form.Label>
                <Form.Control type="email" placeholder="Enter email" />
              </Form.Group>
            </Form.Row>
            <Form.Row>
              <Form.Group as={Col} controlId="usernameSignUp" {...usernameSignUp}>
                <Form.Label>Username</Form.Label>
                <Form.Control placeholder="Enter username" />
              </Form.Group>
            </Form.Row>


            <span>{stateSignUp}</span>

            <Form.Group id="formGridCheckbox">
              <Form.Check type="checkbox" label="Check me out" />
            </Form.Group>

            <Button variant="primary" id='signup-btn' onClick={handleClick}>
              Submit
            </Button>
          </Form>
          </Col>
          <Col></Col>
        </Row>
      </Container>
    </div>

  )
}

function useFormInput(initial) {
  const [value, setValue] = useState(initial)

  function handleChange(e) {
    setValue(e.target.value)
  }
  return {
    value,
    onChange : handleChange
  }
}

export default SignUp
