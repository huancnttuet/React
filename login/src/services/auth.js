import axios from 'axios'

const login = (params) => {
  return axios.post(`${process.env.REACT_APP_API_URL}/signin`, params);
}

const signup = (params) => {
  return axios.post(`${process.env.REACT_APP_API_URL}/signup`, params);
}

const changepwd = (params) => {
  return axios.post(`${process.env.REACT_APP_API_URL}/changepwd`, params);
}

const forgottenacc = (params) => {
  return axios.post(`${process.env.REACT_APP_API_URL}/forgottenacc`, params);
}

export default {
  login,
  signup,
  changepwd,
  forgottenacc,
}
