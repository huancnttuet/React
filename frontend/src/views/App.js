import React from 'react'
import '../components/css/App.css'

import { BrowserRouter as Router, Route } from 'react-router-dom'
import Home from './Home'
import SignIn from '../components/account/SignIn'
import SignUp from '../components/account/SignUp'
import ChangePwd from '../components/account/ChangePwd'
import ForgottenAcc from '../components/account/ForgottenAcc'
import ListTour from '../components/tour/ListTour'
import DetailTour from '../components/tour/DetailTour'
import TopPage from './TopPage'
import Content from '../components/tour/Content'
import Weather from '../components/weather/Weather'
import Manager from '../components/admin/Manager'
import ManagerTour from '../components/admin/ManagerTour'
import AddImage from '../components/admin/AddImage'
import FindTour from '../components/tour/FindTour'
import ManagerOrder from '../components/admin/ManagerOrder'
import OrderDetail from '../components/admin/order/OrderDetail'
import { connect } from 'react-redux'

function App(props) {
	var pathChangePwd = `/changepwd/${props.id}`
	return (
		<Router>
			<div style={{ fontSize: '22px' }}>
				<div>
					<TopPage type={props.authenticate} level={props.level} />
					<Route path='/weather' component={Weather} />
					<Route path='/signin' component={SignIn} />
					<Route path='/signup' component={SignUp} />
					<Route path={pathChangePwd} component={ChangePwd} />
					<Route path='/forgottenacc' component={ForgottenAcc} />

					<Route path='/home' component={Content} />
					<Route path='/tour/:id' component={ListTour} />
					<Route path='/detail/:id' component={DetailTour} />
					<Route path='/find/:data' component={FindTour} />

					<Route path='/manager' component={Manager} />
					<Route path='/diadiem/:id' component={ManagerTour} />
					<Route path='/img/:id' component={AddImage} />
					<Route path='/order' component={ManagerOrder} />
					<Route path='/orderdetail/:id' component={OrderDetail} />
				</div>
			</div>
		</Router>
	)
}

const mapStateToProps = (state) => ({
	id: state.id,
	authenticate: state.authenticate,
	level: state.level
})

export default connect(mapStateToProps)(App)
