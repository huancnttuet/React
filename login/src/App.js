import React from 'react';
import './css/App.css';

import TopPage from './TopPage'
import { BrowserRouter as Router, Route, Link } from "react-router-dom";
import Home from './Home'
import SignIn from './SignIn'
import SignUp from './SignUp'

function App() {
  return (
    <Router>
       <div>
         <TopPage />

         <div >
           <Route exact path="/" component={Home} />
           <Route path="/signin/:id" component={SignIn} />
           <Route path="/signup" component={SignUp} />
         </div>
       </div>


     </Router>
  );
}

export default App;
