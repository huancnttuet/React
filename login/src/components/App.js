import React, {useReducer,  setGlobal} from 'reactn';
import './css/App.css';

import { BrowserRouter as Router, Route } from "react-router-dom";
import Home from './Home'
import SignIn from './SignIn'
import SignUp from './SignUp'
import ChangePwd from './ChangePwd'
import ForgottenAcc from './ForgottenAcc'
import NotFound from './NotFound'
import TopPage from './TopPage'

const reducer = (state, action) => {
  switch (action.type) {
    case 'LOGIN':
      localStorage.setItem('authenticate', true)
      localStorage.setItem('id', action.payload)
      return {authenticate: true, id: action.payload}
    case 'LOGOUT':
      localStorage.setItem('authenticate', false)
      localStorage.setItem('id', 0)
      return {authenticate: false, id: 0}
    default:
      return state
  }
}


function App(props) {

  const [state, dispatch] = useReducer(reducer, {authenticate:localStorage.getItem('authenticate'), id: localStorage.getItem('id')})
  setGlobal({
    state: state,
    dispatch: dispatch
  })

  var pathHome
  if(localStorage.getItem('id') === 0){
    pathHome = '/'
  } else {
    pathHome = `/${localStorage.getItem('id')}`
  }

  console.log(pathHome);
  return (
    <Router>
        <TopPage />
         <div >
          <Route exact path={pathHome} component={Home} />
          <Route exact path='/' component={() => <SignIn authenticate={localStorage.getItem('authenticate')} /> } />
          <Route path="/signup" component={SignUp} />
          <Route path='/changepwd' component={ChangePwd} />
          <Route path="/forgottenacc" component={ForgottenAcc} />
         </div>

     </Router>
  );
}


export default App;
