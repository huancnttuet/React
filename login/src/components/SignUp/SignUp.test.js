import React from "react";
import { shallow, mount } from "enzyme";
import App from "../App";
import SignUp from "./SignUp";
import { authServices } from "services";


let wrapper;
beforeEach(() => {
  wrapper = mount(<App />);
});

afterEach(() => {
  wrapper.unmount();
});

describe("SignUp", () => {
  it("click button signup", () => {
    const loginMock = jest.spyOn(authServices, "signup");
    const wrapper1 = mount(<SignUp />);
    expect(wrapper1.debug()).toMatchSnapshot();
    wrapper1.find("#emailSignUp").simulate("change", {
      target: {
        value: "arsenal"
      }
    });
    wrapper1.find("#usernameSignUp").simulate("change", {
      target: {
        value: "chelsea"
      }
    });
    wrapper1
      .find("#signup-btn")
      .at(0)
      .simulate("click");
    wrapper1
      .find("#signup-btn")
      .at(1)
      .simulate("click");
    expect(loginMock).toHaveBeenCalled();
  });
  test("test API", () => {
    const data = {
      usernameSignUp: `arsenal`,
      emailSignUp: `chelsea`
    };
    return authServices.signup(data).then(data => {
      expect(data.data).toMatchObject({ code: "EXIST" });
    });
  });
  //
});
