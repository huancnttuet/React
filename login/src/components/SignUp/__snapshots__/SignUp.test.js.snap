// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUp click button signup 1`] = `
"<SignUp>
  <div>
    <Container as=\\"div\\" fluid={false}>
      <div className=\\"container\\">
        <Bootstrap(Row) style={{...}}>
          <Row style={{...}} bsPrefix=\\"row\\" as=\\"div\\" noGutters={false}>
            <div style={{...}} className=\\"row\\">
              <Col as=\\"div\\">
                <div className=\\"col\\" />
              </Col>
              <Col as=\\"div\\">
                <div className=\\"col\\">
                  <Form inline={false} as=\\"form\\">
                    <form className=\\"\\">
                      <FormRow>
                        <div className=\\"form-row\\">
                          <FormGroup as={{...}} controlId=\\"emailSignUp\\" value=\\"\\" onChange={[Function: handleChange]}>
                            <Col value=\\"\\" onChange={[Function: handleChange]} className=\\"form-group\\" as=\\"div\\">
                              <div value=\\"\\" onChange={[Function: handleChange]} className=\\"form-group col\\">
                                <FormLabel column={false} srOnly={false}>
                                  <label htmlFor=\\"emailSignUp\\" className=\\"form-label\\">
                                    Email
                                  </label>
                                </FormLabel>
                                <FormControl type=\\"email\\" placeholder=\\"Enter email\\" as=\\"input\\">
                                  <input placeholder=\\"Enter email\\" type=\\"email\\" readOnly={[undefined]} id=\\"emailSignUp\\" className=\\"form-control\\" />
                                </FormControl>
                              </div>
                            </Col>
                          </FormGroup>
                        </div>
                      </FormRow>
                      <FormRow>
                        <div className=\\"form-row\\">
                          <FormGroup as={{...}} controlId=\\"usernameSignUp\\" value=\\"\\" onChange={[Function: handleChange]}>
                            <Col value=\\"\\" onChange={[Function: handleChange]} className=\\"form-group\\" as=\\"div\\">
                              <div value=\\"\\" onChange={[Function: handleChange]} className=\\"form-group col\\">
                                <FormLabel column={false} srOnly={false}>
                                  <label htmlFor=\\"usernameSignUp\\" className=\\"form-label\\">
                                    Username
                                  </label>
                                </FormLabel>
                                <FormControl placeholder=\\"Enter username\\" as=\\"input\\">
                                  <input placeholder=\\"Enter username\\" type={[undefined]} readOnly={[undefined]} id=\\"usernameSignUp\\" className=\\"form-control\\" />
                                </FormControl>
                              </div>
                            </Col>
                          </FormGroup>
                        </div>
                      </FormRow>
                      <span />
                      <FormGroup id=\\"formGridCheckbox\\" as=\\"div\\">
                        <div id=\\"formGridCheckbox\\" className=\\"form-group\\">
                          <FormCheck type=\\"checkbox\\" label=\\"Check me out\\" inline={false} disabled={false} isValid={false} isInvalid={false} title=\\"\\">
                            <div style={[undefined]} className=\\"form-check\\">
                              <FormCheckInput type=\\"checkbox\\" isValid={false} isInvalid={false} isStatic={false} disabled={false}>
                                <input type=\\"checkbox\\" disabled={false} id={[undefined]} className=\\"form-check-input\\" />
                              </FormCheckInput>
                              <FormCheckLabel title=\\"\\" type=\\"checkbox\\">
                                <label title=\\"\\" type=\\"checkbox\\" htmlFor={[undefined]} className=\\"form-check-label\\">
                                  Check me out
                                </label>
                              </FormCheckLabel>
                            </div>
                          </FormCheck>
                        </div>
                      </FormGroup>
                      <Button variant=\\"primary\\" id=\\"signup-btn\\" onClick={[Function: handleClick]} active={false} disabled={false} type=\\"button\\">
                        <button id=\\"signup-btn\\" onClick={[Function: handleClick]} disabled={false} type=\\"button\\" className=\\"btn btn-primary\\">
                          Submit
                        </button>
                      </Button>
                    </form>
                  </Form>
                </div>
              </Col>
              <Col as=\\"div\\">
                <div className=\\"col\\" />
              </Col>
            </div>
          </Row>
        </Bootstrap(Row)>
      </div>
    </Container>
  </div>
</SignUp>"
`;
